src/price_data.py

```diff
@@ -326,3 +328,133 @@ def get_cost(self, tr: Union[Operation, SoldCoin], reference_coin: str = config.
         if isinstance(tr, SoldCoin):
             return price * tr.sold
         raise NotImplementedError
+    
+    def get_candles(self, start: int, stop: int, symbol: str) ->list:
+        if self.exchange.has['fetchOHLCV']:
+            sleep(self.exchange.rateLimit / 1000)  # time.sleep wants seconds
+            # get 2min before and after range
+            return self.exchange.fetch_ohlcv(symbol, '1m', start-1000*60*2, max(int((stop-start)/1000/60)+5, 1))
+        else:
+            logging.warning(
+                "fetchOHLCV not implemented on exchange, skipping priceloading using ohlcv")
+            raise None
+
+    def initialize_ccxt(self):
+        exchange_id = 'binance'
+        exchange_class = getattr(ccxt, exchange_id)
+        self.exchange = exchange_class()
+        self.markets = []
+        markets = self.exchange.fetch_markets()
+
+        for market in markets:
+            # may not apply for all exchanges, currently works for binance
+            # caches a list of all pairs on the exchange
+            self.markets.append(market["symbol"].split("/"))
+
+    def _get_binance_bulk_pair_list(self, reference_coin: str = config.FIAT, coin) -> list:
```

Comment:
1. missed that when adding type hints^^
2. yes this should work for all. ill rename it
3. yeah that would be nice. What i wrote was a more simple solution as a starting point, because i wanted a working version before starting fancier stuff. Also as a problem if we have muliple routes of the same length how do we determine the best route (for instance always picking going over more stable coins/fiat like USDT or how its called on the exchange, to reduce error)

Reply:
Just noticed :
Just having a pair doesn't mean that it was active a the requested candle on that exchange. The current method only gets every pair that was ever active on that exchange to my knowledge but not the time-frame it was active). We'd need to implement a fallback to other exchanges if we cant fetch from the desired one (would be eliminated if we could find out using ccxt from when to when a pair was active on a specific exchange.)

this would also fix the problem i had with IOTA/ETH  and ETH/EUR on binance

This is more complicated than i thought it would be^^

Reply:
3. definitly have to think about that. We might even group different transactions which happend in the same minute. And save all retrieved prices. So not just DODGE/EUR, if we went over DODGE/BTC, BTC/EUR. So we could reuse them.

Reply:
this is also obsolete due to the graph approach. I currently do not cache candle data that could be an area of improvement

it currently does not use a good path finding algorithm which can get slow for large amount of pairs but for now its fast enough